AsyncDisplayKit/ASMapNode.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+#import <MapKit/MapKit.h>
+@interface ASMapNode : ASControlNode
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate NS_DESIGNATED_INITIALIZER;
+
+@property (nonatomic, readonly) ASImageNode *mapImage;
+@property (nonatomic, readonly) ASDisplayNode *liveMap;
+/**
+ Whether the map snapshot should turn into a MKMapView when tapped on. Defaults to YES.
+ */
+@property (nonatomic, assign) BOOL hasLiveMap;
```

Comment:
Yeah that sounds good, I will change it to default to NO. I prefer changing it to becomeLiveMap since that makes it clear that it's an action at that point rather than it becoming a live map later on. This would make it clear that it would show/hide depending on the developer setting YES/NO. That way we can remove any internal actions so that the developer is free to handle the touch in anyway they choose (they may want to open a separate view to see the map in full screen) making it much more flexible.